lib/puppet/util/monkey_patches.rb

```diff
@@ -20,6 +20,9 @@ def self.caller(skip=nil)
 
 class Symbol
   def <=> (other)
+    if (other.class != Symbol)
```

Comment:
I think there was some concern that this comparison itself could be expensive if there were a lot of them. I don't think this concern warrants a standing benchmark but maybe a one-off test to see if there could be performance implications.